File: reactui/src/common/models/building.ts
import { IBuilder } from "./builder";

export interface IBuilding {
    buildingId: string;
    builder: IBuilder | null;
    type: string;
    amount: number;
    woodworkThisYear: number;
    stoneworkThisYear: number;
    smithsworkThisYear: number;
    woodThisYear: number;
    stoneThisYear: number;
    ironThisYear: number;
    upkeep: number;
    woodwork: number;
    stonework: number;
    smithswork: number;
    wood: number;
    stone: number;
    iron: number;
}

File: Domain/Entities/Building.cs
using System;
using Domain.Entities.Persons;
using Domain.Entities.Types;

namespace Domain.Entities
{
    public class Building
    {
        public Guid BuildingId { get; set; }
#nullable enable
        public virtual Builder? Builder { get; set; }
#nullable disable
        public virtual Fief Fief { get; set; }
        public virtual BuildingType BuildingType { get; set; }
        public string Type { get; set; }
        public int Amount { get; set; }
        public int WoodworkThisYear { get; set; }
        public int StoneworkThisYear { get; set; }
        public int SmithsworkThisYear { get; set; }
        public int WoodThisYear { get; set; }
        public int StoneThisYear { get; set; }
        public int IronThisYear { get; set; }
    }
}